arch/arm/src/stm32/stm32_i2c_v2.c

```diff
@@ -2144,6 +2144,25 @@ static int stm32_i2c_init(struct stm32_i2c_priv_s *priv)
   modifyreg32(STM32_RCC_APB1RSTR, 0, priv->config->reset_bit);
   modifyreg32(STM32_RCC_APB1RSTR, priv->config->reset_bit, 0);
 
+  /* Configure clock source to HSI */
```

Comment:
this should go to stm32g4xxxx_rcc.c file where other CCIPR related config is done